### PR TITLE
[DOCS] Reformat `shingle` token filter

### PR DESCRIPTION
**Pull request by jrodewig**
Changes:

* Rewrites description and adds Lucene link
* Adds analyze example
* Rewrites parameter documentation
* Updates custom analyzer and filter examples
* Adds anchor to `index.max_shingle_diff` index-level setting

### CODE REVIEW

File: docs/reference/analysis/tokenfilters/shingle-tokenfilter.asciidoc [hunk 1]
```diff
@@ -4,46 +4,507 @@
...
-at which there is no actual token in the stream. For instance this string is
-used if the position increment is greater than one when a `stop` filter is used
-together with the `shingle` filter. Defaults to `"_"`
-|=======================================================================
-
-The index level setting `index.max_shingle_diff` controls the maximum allowed
-difference between `max_shingle_size` and `min_shingle_size`.
+Add shingles, or word https://en.wikipedia.org/wiki/N-gram[n-grams], to a token
+stream by concatenating adjacent tokens. By default, the `shingle` token filter
+creates two-word shingles.
```
Comment (mayya-sharipova) [inline docs/reference/analysis/tokenfilters/shingle-tokenfilter.asciidoc:9]: should we also say "two-word shingles and unigrams"?

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig Thanks! This looks very good to me.

File: docs/reference/analysis/tokenfilters/shingle-tokenfilter.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/analysis/tokenfilters/shingle-tokenfilter.asciidoc]: Good catch. I've updated this to "outputs two-word shingles and unigrams".

**Comment by jrodewig** [comment]
Thanks @mayya-sharipova

**Comment by jrodewig** [comment]
### Backport commits
master 00ab16f 
7.x a2de43d 
7.8 85da75d
7.7 902f0cd

**Pull request merged**